cpp/modmesh/python/wrapper/modmesh/wrap_SimpleArray.cpp

```diff
@@ -163,99 +150,308 @@ class MODMESH_PYTHON_WRAPPER_VISIBILITY WrapSimpleArray
+                ellipsis_flag = true;
+                break;
+            }
+
+            auto & slice_out = slices[it - tuple.begin()];
+            const auto slice_in = (*it).cast<py::slice>();
+
+            copy_slice(slice_out, slice_in);
+        }
+
+        // copy slices from the back until an ellipsis
+        if (ellipsis_flag)
+        {
+            for (size_t size = 0; size < tuple.size(); size++)
+            {
+                auto it = tuple.end() - size - 1;
+
+                if (py::isinstance<py::ellipsis>(*it))
+                {
+                    break;
+                }
+                auto & slice_out = slices[ndim - size - 1];
+                const auto slice_in = (*it).cast<py::slice>();
+
+                copy_slice(slice_out, slice_in);
+            }
+        }
+    }
+
+    static void slice_syntax_check(pybind11::tuple const & tuple, size_t ndim)
+    {
+        namespace py = pybind11;
+
+        size_t ellipsis_cnt = 0;
+        size_t slice_cnt = 0;
+
+        for (auto it = tuple.begin(); it != tuple.end(); it++)
+        {
+            if (py::isinstance<py::ellipsis>(*it))
+            {
+                ellipsis_cnt += 1;
+            }
+            else if (py::isinstance<py::slice>(*it))
+            {
+                slice_cnt += 1;
+            }
+            else
+            {
+                throw std::runtime_error("unsupported operation.");
+            }
+        }
+
+        if (ellipsis_cnt + slice_cnt > ndim)
+        {
+            throw std::runtime_error("syntax error. dimensions mismatches");
+        }
+
+        if (ellipsis_cnt > 1)
+        {
+            throw std::runtime_error("syntax error. no more than one ellipsis.");
+        }
+    }
+
+    static void broadcast_array_using_slice(wrapped_type & arr_out,
+                                            std::vector<small_vector<int>> const & slices,
+                                            pybind11::array const & arr_in)
+    {
+        TypeBroadCast::check_shape(arr_out, slices, arr_in);
+
+        size_t nghost = 0;
+
+        if (arr_out.has_ghost())
+        {
+            nghost = arr_out.nghost();
+            arr_out.set_nghost(0);
+        }
+
+        TypeBroadCast::broadcast(arr_out, slices, arr_in);
+
+        if (nghost != 0)
+        {
+            arr_out.set_nghost(nghost);
+        }
+    }
+
     struct TypeBroadCast
     {
-        static void check_shape(wrapped_type const & arr_out, pybind11::array const & arr_in)
+        static void check_shape(wrapped_type const & arr_out,
+                                std::vector<small_vector<int>> const & slices,
+                                pybind11::array const & arr_in)
         {
-            const auto & left_shape = arr_out.shape();
-            const auto * right_shape = arr_in.shape();
+            shape_type right_shape(arr_in.ndim());
+            for (pybind11::ssize_t i = 0; i < arr_in.ndim(); i++)
```

Comment:
The slice handling code here is a good staring point for adding view support for `SimpleArray` in the future.

Reply:
I will need to discuss this with you later.

Reply:
We can use dc or https://github.com/solvcon/modmesh/discussions to discuss.